### PR TITLE
[Semantic Segmentation] Add CamvidCustom dataset class.

### PR DESCRIPTION
**Pull request by iizukak**
## Motivation and Context

This PR is part of #88
Based on #189 

## Description

- Implement `CamvidBase` class
- Implement `CamvidCustom` class
- Make `Cmavid` class inhert `CamvidBase` class

`CamvidCustom` class has `labels.txt`.
`labels.txt` can include `Ignore` label. `Ignore` label should be ignore when training. This PR also include `Ignore` label test.

## How has this been tested?

Write unit test for `Camvid` and `CamvidCustom.`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/datasets/camvid.py [hunk 1]
```diff
@@ -37,9 +38,85 @@ def get_image(filename, convert_rgb=True):
...
+    https://github.com/alexgkendall/SegNet-Tutorial/tree/master/CamVid
+    """
+    def __init__(
+            self,
+            batch_size=10,
+            *args,
+            **kwargs
+    ):
+
+        self.use_prefetch = kwargs.pop("enable_prefetch", False)
```
Comment (ruimashita) [inline lmnet/lmnet/datasets/camvid.py]: No need this line anymore.

File: lmnet/lmnet/datasets/camvid.py [hunk 2]
```diff
@@ -37,9 +38,85 @@ def get_image(filename, convert_rgb=True):
...
+        self.num_prefetch_process = kwargs.pop("num_prefetch_processes", 8)
```
Comment (ruimashita) [inline lmnet/lmnet/datasets/camvid.py]: No need this line anymore.

File: lmnet/lmnet/datasets/camvid.py [hunk 3]
```diff
@@ -37,9 +38,85 @@ def get_image(filename, convert_rgb=True):
...
+
+        super().__init__(
+            batch_size=batch_size,
+            *args,
+            **kwargs,
+        )
+
+    extend_dir = "CamVid"
+
+    def prefetch_args(self, i):
```
Comment (ruimashita) [inline lmnet/lmnet/datasets/camvid.py]: No need this function anymore.

File: lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    train_dataset = DummyCamvidCustomWithoutTestDataset(subset="train", batch_size=batch_size,
+                                                        validation_size=validation_size)
+    train_dataset = DatasetIterator(train_dataset)
+
+    test_dataset = DummyCamvidCustomWithoutTestDataset(subset="validation", batch_size=batch_size,
+                                                       validation_size=validation_size)
+    test_dataset = DatasetIterator(test_dataset)
+
+    image_files, label_files = train_dataset.feed()
```
Comment (ruimashita) [inline lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py]: It is better to add test that train and validation data number is separated correctly like this  
https://github.com/blue-oil/blueoil/blob/master/lmnet/tests/lmnet_tests/datasets_tests/test_open_images_v4.py#L124-L125

File: lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    label = PIL.Image.fromarray(label_rgb)
+
+    image.show()
+    label.show()
+
+
+def _test_camvid_basics(train_dataset, test_dataset):
+    assert train_dataset.num_classes == 11
+    colors = train_dataset.label_colors
+    assert len(colors) == 12
```
[the remark this answers is not in the document]
Comment (iizukak) [reply lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py]: Yes. +1 color is for background (unlabeled).

**Comment by iizukak** [comment]
@ruimashita 
I removed needless lines and added test for splited dataset size.

**Comment by ruimashita** [comment]
@iizukak Thanks 👍

File: lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ruimashita) [reply lmnet/tests/lmnet_tests/datasets_tests/test_camvid.py]: Currently the model assume ignore background class, but it should be able to choose by user, I think.

**Comment by iizukak** [comment]
@ruimashita Review Please.

This PR include #189. So I want to merge #189 first.

**Comment by iizukak** [comment]
I replace Ignored class pixel with 255...
I want better way.

File: lmnet/lmnet/datasets/camvid.py [hunk 6]
```diff
@@ -140,15 +201,94 @@ def files_and_annotations(self):
...
+            batch_size=batch_size,
+            *args,
+            **kwargs,
+        )
+        self.parse_labels()
+
+    @property
+    def label_colors(self):
+        classes = self.parse_labels()
+        return COLOR_MAP[:len(classes)]
```
Comment (ruimashita) [inline lmnet/lmnet/datasets/camvid.py]: If the length of class is over than 21, the label_colors() return 20 length list of colors, I think. Is the length of colors what you expect ??

Comment (iizukak) [reply lmnet/lmnet/datasets/camvid.py]: @ruimashita Oh, this is my lack of consideration.
I added `get_color_map` function to get arbitrary length color map.
I also added unit test in `test_common.py`

Thanks!!
